src/Compilers/CSharp/Test/Semantic/Semantics/RecordTests.cs

```diff
@@ -1506,6 +1506,9 @@ public static void Main()
 ";
             var comp = CreateCompilation(src);
             comp.VerifyDiagnostics(
+                // (5,25): warning CS0067: The event 'C.X' is never used
```

Comment:
This is just wrong? It's used.

Reply:
My impression is that just assigning a field-like event is not sufficient to remove this warning. See for example

```cs
using System;
class C
{
    public event Action X; // warning: the event 'C.X' is never used
    public static void Main()
    {
        var c = new C();
        c.X += () => {};
    }
}
```

The fact that my PR *introduces* the warning is interesting. I am assuming it is a side effect of removing the UnassignedFieldsWalker.

Reply:
OK, some of the existing behaviors in master are mystifying me.

[This](https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQBMQGoA+ABATARgFgAoPAAgGESBvEs+s7AZjLgDc4A7GR/XMgBoBuMgHpRZAO4QEnAJacA5iDIwAFnDIAzOXAA2KMgHIKAOgFGycqGWhQ5iznEPAArjzkwbbCHtebrMicOBDJXKGc6BjxcKPpaYgYksh9QgGMyAF4guElKAAoASiE45LTzLKDXPT0SxIYAXxIGoA===) warns, which seems good:
```cs
using System;
class C
{
    public event Action X; // warning: the field 'C.X' is assigned but its value is never used
    public static void Main()
    {
        var c = new C();
        c.X = null;
    }
}
```

[This](https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQBMQGoA+ABATARgFgAoPAAgGESBvEs+s7AZjLgDc4A7GR/XMgBoBuMgHpRZTgHsyAdwgJOAS04BzAPx0GeXFvq1iDI2TYKyAYzIBeSXFmUAFAEohe4+YB0A62WfWAfGTUAL6uhgzBJMFAA=) does not warn. What?
```cs
using System;
class C
{
    public event Action X; // no warning?
    public static void Main()
    {
        var c = new C();
        c.X = () => {};
    }
}
```

[This](https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQBMQGoA+ABATARgFgAoPAAgGESBvEs+s7AZjLgDc4A7GR/XMgBoBuMgHpRZTgHsyAdwgJOAS04BzAPx0GzXgDZGAFjIBZfAApsfMhACUW+rWINnZNgrIBjMgF5JcWZRmNkL2Lh4AdAI+1iFODAC+JPFAA==) also does not warn. So I don't think it's related to some kind of capturing bug..?
```cs
using System;
class C
{
    public event Action X; // no warning?
    public static void M1(Action a)
    {
        var c = new C();
        c.X = a;
    }
}
```

Reply:
Probably worth an email discussion on what the correct behavior is.

---
In reply to: [470185248](https://github.com/dotnet/roslyn/pull/46441#discussion_r470185248) [](ancestors = 470185248)

Reply:
Happy to follow up and examine these behaviors more closely in an email thread. I will leave these baselines as they currently stand in this PR.